core/include/macros/units.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2020 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     core_macros
+ * @{
+ *
+ * @file
+ * @brief       Unit helper macros
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifndef UNITS_H
+#define UNITS_H
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+/* KiB, MiB, GiB */
+#define KiB(x) ((unsigned long long)(x) << 10)
+#define MiB(x) (KiB(x) << 10)
+#define GiB(x) (MiB(x) << 10)
+
+/* kHz, MHz */
+#define KHZ(x)    ((x) * 1000UL)
+#define MHZ(x) (KHZ(x) * 1000UL)
```

Comment:
Well, so as is, this works up to 4192MHz. Same as above. Is that sufficient? Does the compiler create inefficient code when used with small values, should this be cast to ULL?

Reply:
I think 4 GHz should be enough for everything conceivable (for radios operating at [higher frequencies](https://www.decawave.com/product/dw1000-radio-ic/) it is very unlikely that we'd have to pass around the center frequency as one large 64-bit integer)

But I can change it to ULL if you prefer.

Reply:
Well, does the compiler warn if `MHZ(5123)` is used? that's a valid 802.11n frequency, users could use that.

Reply:
I think most drivers would use channels - anyway, changed it just to ULL just to be sure.
For a [quick check](https://godbolt.org/z/uLjhcX) this did not affect the generated code.

Reply:
> For a [quick check](https://godbolt.org/z/uLjhcX) this did not affect the generated code.

nice, thanks for checking!